modules/data_release/ajax/FileUpload.php

```diff
@@ -17,33 +17,32 @@
 $user    = $factory->user();
 $config  = $factory->config();
 
-if ($_POST['action'] == 'upload'
-    && $user->hasPermission("data_release_upload")
-) {
+if (!$user->hasPermission('data_release_upload')) {
+    header("HTTP/1.1 403 Forbidden");
+    exit;
+}
+
+if ($_GET['action'] == 'upload') {
     $fileName    = $_FILES["file"]["name"];
-    $version     = $_POST['version'];
+    $version     = !empty($_POST['version']) ? $_POST['version'] : null;
     $upload_date = date('Y-m-d');
 
     $settings = $factory->settings();
 
     $baseURL = $settings->getBaseURL();
     $path    = \Utility::appendForwardSlash($config->getSetting('dataReleasePath'));
-
     if (!file_exists($path)) {
-        error_log(
-            "ERROR: File upload failed. Upload"
-            . " directory not found."
-        );
-        header("HTTP/1.1 500 Internal Server Error");
+        $msg = "File upload failed. Upload directory not found.";
+        error_log('ERROR: ' . $msg);
+        header("HTTP/1.1 404 $msg");
     } elseif (!is_writable($path)) {
-        error_log(
-            "File upload failed. Upload directory"
-            . " does not appear to be writeable."
-        );
-        header("HTTP/1.1 500 Internal Server Error");
+        $msg = "File upload failed. Upload directory is not writeable.";
+        error_log('ERROR: ' . $msg);
+        header("HTTP/1.1 404 $msg");
```

Comment:
same. The old 500 Internal Server Error also seems to be a more accurate response.

Reply:
updated in the last commit